### DISCUSSION TITLE
Home Assistant and ebusd

### DISCUSSION THREAD

**Discussion by rbaleja** [john30/ebusd#1230] [General]
I'm installing HA and ebusd for the first time. I have a question whether the ebusd daemon should work in HA OS? Is only the ebus-addon HA plugin enough? If daamon is to work, how to install it on HA? For now, I've run ebusd on raspberry and I'm reading data using ebusctl, but I don't know how to integrate it with HA.

**Comment by JonesGW-MGD** [comment]
Hi Robert, the ebusd add-on (LukasGrebe and others) isn't the only way to run the ebusd daemon with HA but it's probably the easiest to set up.  You'll need that add-on, Mosquitto MQTT add-on and of course an ebus hardware adapter - but that's it.  

Otherwise you'll need to do the integration yourself.  I'm no expert but HAOS seems to use some form of BalenaOS/ResinOS that gives the supervisor/add-on architecture by running these in containers and isn't your usual RasPi linux distribution.  You'll probably end up with your ebusd on 1 Pi and another running HAOS I would imagine?  Or adding your own container to the HA Pi but then you're essentially replicating some of the add-on capability without the benefits of seeing/controlling the daemon from HA

**Reply by martinjowebb** [reply]
Sorry to hijack this a bit but how did you get the ebus addon installed on HAOS? Ive added it to my configuration.yaml (below) but can't see any evidence that its running

ebusd:
  host: 192.168.0.70
  circuit: "bai"
  port: 9999

**Reply by Koky05** [reply]
In HAOS go to Addon store and add custom repository `https://github.com/LukasGrebe/ha-addons/`
Then you can install eBusd deamon addon. But if you already running ebus deamon then you just configure it send date into MQTT broker with HA Autodiscovery, detail in wiki [MQTT Client](https://github.com/john30/ebusd/wiki/3.3.-MQTT-client).
It is hard to explain in more detail because we don't know how do you running ebus deamon.

**Reply by martinjowebb** [reply]
Thanks so much. Have been searching for hours for that one piece of information that you need to install a custom repo to see the addon! Now can see all the data from my boiler

**Comment by rbaleja** [comment]
Hi. Thank you for the information. I understand that if I have the plugin (https://github.com/LukasGrebe/ha-addons/), I no longer need ebusd (daemon). I connect raspi with standardly installed HA and (in my case an ebusd adapter with a LAN interface). Of course, I configure the plugin according to the documentation...ens: etc. And the next step is MQTT. If this is the case, my mistake was various attempts to install ebusd daemon on HAOS or HA Supervisor. Which of course didn't work.

**Reply by Koky05** [reply]
I will switch steps. Install MQTT addon and configure it to work with HA. Then configure and connect eBUS because HA eBus Addon will need information about your MQTT broker installed in first step.

**Reply by rbaleja** [reply]
I'll test this one and let you know after some time. Thank you for your time. Best regards. Robert SP7RB

**Reply by JonesGW-MGD** [reply]
I'd concur with the order that @Koky05 is proposing.

it's not that you don't need ebusd daemon, it's just that the add-on provides a 'wrapper' to the ebusd code to conform to the HA add-on specs - and thus once installed it runs ebusd in a container, and gives you some of the ebusd options configurable in the add-on configuration in more of a gui-format.

As you'll see from this page: https://github.com/LukasGrebe/ha-addons/blob/main/ebusd/DOCS.md: 

_"Deep dive into the [ebusd Wiki](https://github.com/john30/ebusd/wiki). **The steps Build and Install** (my emphasis) are handled by this addon. You'll need to configure the options in this addon to run ebusd."_

**Reply by rbaleja** [reply]
So do I have to install ebusd myself or will the add-on do it? How to do it yourself in a standard HA installation? Unless the Ha installation is to be different. I have read the documentation and I can compile or download the installer from the repo, but only on Rasp with Debian. Unless it's supposed to work like that. one Raspi with ebusd, the other with HA, but then probably without a plugin because communication via MQTT? If there is to be ebusd in the container, I have to install HA differently.

**Reply by Koky05** [reply]
You need only one ebus deamon, but you can decised if it will be as standalone container on any device or as addon in HAOS. Every ebusd will communicate with HA via MQTT.

**Reply by rbaleja** [reply]
So good to understand. So the Home Assistant Add-on: eBUSd plugin is a replacement for ebusd in HA? The documentation that talks about compiling and installing ebusd is confusing.

**Reply by JonesGW-MGD** [reply]
More or less.  The add-on **is** ebusd.  (you don't need to separately install ebusd - just follow the add-on instructions). It is a package that builds and installs ebusd, and containerises it, suiting HA's add-on architecture.  The standard ebusd instructions on the other hand are non-HA specific - that it is to say, if you want to implement ebusd without HA - which many people do - standalone or openHAB users (or if you have HA but want to separately install ebusd on say another pi etc.) for example.  In these cases you would have to build and install ebusd yourself.

For your needs - Start with the add-on instructions and as it says - understand first what ebusd does by reference to the ebusd wiki, (and how it does it if you are interested - the documentation is comprehensive) and then, understanding that you don't need to do the build and install part, follow the instructions for the add-on.  Once you have it up and running, revert to the ebusd wiki to better understand how ebusd needs to be configured to optimise your set-up.

Generally you will find that, once working, ebusd will be publishing data about your set-up to MQTT and HA will be consuming this and will automatically create sensors that will have relevant data.

You should therefore find a wealth of data 'out-of-the-box' once it is running correctly.  

You may also find that not all the datapoints you may want are available -  that is when it becomes a 'project' for you to determine whether there may be some tweaking of the message configuration files to better suit your equipment.  (Typically and as you'd expect, the message definitions on ebus aren't readily available from Vaillant and other manufacturers - so what you have access to is the ebusd community's best understanding of the various messages.  Newer equipment is likely to be less well documented than older equipment, and so on.)

It then becomes a 'time spent v reward' exercise for you to decide how involved you get!

So first - go here:  https://github.com/LukasGrebe/ha-addons#how-to-install

**Reply by rbaleja** [reply]
It finally worked with the HA plug-in. There was a stupid configuration error ( twice entry ens:ens: in yaml file). I already have data from MQTT broker. Thanks.
